src/snowflake/connector/query_context_cache.py

```diff
@@ -80,7 +66,12 @@ def __lt__(self, other: object) -> bool:
             raise NotImplementedError(
                 f"cannot compare QueryContextElement with object of type {type(other)}"
             )
-        return self._priority < other.priority
+        if self._priority != other._priority:
+            return self._priority < other._priority
+        if self._id != other._id:
+            return self._id < other._id
+        return self._read_timestamp < other._read_timestamp
```

Comment:
> this aims to fix a bug.

This is the exact reason why I'd like to see it in a separate PR because we should not have one PR do a hundred of things. If this PR's title is SerDe, we should only have SerDe related changes. Having multiple stuff in one PR just increase the possibility of it getting reverted and make debug harder.

Given this does not go into main, I'm fine either way, but just pointing out that fixing a bug is not a good reason to stitch changes.

Reply:
I see. I can revert this part change and make another PR. I just realize that this part change can not fix the multi-database query query context `merge` missing error.